src/pperm.cc

```diff
@@ -3657,67 +3658,16 @@ static Obj FuncOnPosIntSetsPartialPerm(Obj self, Obj set, Obj f)
     RequireSmallList("OnPosIntSetsPartialPerm", set);
     RequirePartialPerm("OnPosIntSetsPartialPerm", f);
 
-    UInt2 *     ptf2;
-    UInt4 *     ptf4;
-    UInt        deg;
-    const Obj * ptset;
-    Obj *       ptres, tmp, res;
-    UInt        i, k, reslen;
+    const UInt len = LEN_LIST(set);
 
-    if (LEN_LIST(set) == 0)
+    if (len == 0)
         return set;
 
-    if (LEN_LIST(set) == 1 && ELM_LIST(set, 1) == INTOBJ_INT(0)) {
+    if (len == 1 && ELM_LIST(set, 1) == INTOBJ_INT(0)) {
         return FuncIMAGE_SET_PPERM(self, f);
     }
 
-    PLAIN_LIST(set);
-    res = NEW_PLIST_WITH_MUTABILITY(IS_PLIST_MUTABLE(set), T_PLIST_CYC_SSORT,
-                                    LEN_PLIST(set));
-
-    /* get the pointer                                                 */
-    ptset = CONST_ADDR_OBJ(set) + LEN_LIST(set);
-    ptres = ADDR_OBJ(res) + 1;
-    reslen = 0;
-
-    if (TNUM_OBJ(f) == T_PPERM2) {
-        ptf2 = ADDR_PPERM2(f);
-        deg = DEG_PPERM2(f);
-        /* loop over the entries of the tuple                              */
-        for (i = LEN_LIST(set); 1 <= i; i--, ptset--) {
-            k = INT_INTOBJ(*ptset);
```

Comment:
Note how this code does not verify that `*ptset` is an immediate integer, nor that it is a *positive* immediate integer. With this PR, we effectively add an `IS_POS_INTOBJ` (by calling `OnSetsPPerm`, which has that check). I imagine that if we execute this loop a couple million times, one might be able to measure this; but I didn't conduct any experiments, as I have no idea what a typical use case would be. Anyway, personally I'd take the safety over speed here any day, but I don't feel it's my decision to make for this code!